crates/biome_js_analyze/src/lint/nursery/use_default_case.rs

```diff
@@ -0,0 +1,195 @@
+use std::ops::Not;
+
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_syntax::{JsLanguage, JsSwitchStatement};
+use biome_rowan::{syntax::SyntaxTrivia, AstNode};
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+declare_rule! {
+    /// Require default cases in switch statements.
+    ///
+    /// See https://eslint.org/docs/latest/rules/default-case
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// switch (a) {
+    ///     case 1:
+    ///         /* code */
+    ///         break;
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// switch (a) {
+    ///     // no default
+    /// 	case 1:
+    ///         break;
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// switch (a) {
+    ///     case 1:
+    ///         break;
+    ///     // no default
+    ///     // nope
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// switch (a) {
+    ///     case 1:
+    ///         /* code */
+    ///         break;
+    ///
+    ///     default:
+    ///         /* code */
+    ///         break;
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// switch (a) {
+    ///     case 1:
+    ///         /* code */
+    ///         break;
+    ///
+    ///     // no default
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// switch (a) {
+    /// }
+    /// ```
+    pub UseDefaultCase {
+        version: "next",
+        name: "useDefaultCase",
+        sources: &[RuleSource::Eslint("default-case")],
+        recommended: false,
+    }
+}
+
+/// Options for the rule `useDefaultCase`.
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct UseDefaultCaseOptions {
+    /// Regular expression that filters the comment pattern to disable the rule.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub comment_pattern: Option<String>,
```

Comment:
I think we should not support `no default` comment because we have already a suppression mechanism based on `biome-ignore` comments. We made the same decision for [noFallthroughSwitchClause](https://biomejs.dev/linter/rules/no-fallthrough-switch-clause/).

Reply:
that makes sense, and make the whole implementation a lot simpler, done